.nuspec/maui.aotprofile.txt

```diff
@@ -2793,9 +2802,6 @@ Methods:
 	System.Reflection.Assembly System.Runtime.Loader.AssemblyLoadContext:MonoResolveUsingResolveSatelliteAssembly (intptr,string)
 	System.Reflection.Assembly System.Runtime.Loader.AssemblyLoadContext:ResolveSatelliteAssembly (System.Reflection.AssemblyName)
 	System.Reflection.Assembly System.RuntimeType:get_Assembly ()
-	System.Reflection.Assembly[] System.AppDomain:GetAssemblies ()
```

Comment:
We used to call this? Nice that was removed!

Reply:
What's interesting is that this call still seems to be on the call path:

https://github.com/dotnet/maui/blob/fe7bc04c54ea94781a44805bad81b0c2548f4c86/src/Controls/src/Core/DependencyService.cs#L109-L119

If you look, we are still calling `void Microsoft.Maui.Controls.DependencyService:Initialize ()` in this AOT profile. And  the Initialize method will call `GetASsemblies()`. So there must have been a change that is setting `s_initialized` to `true` before `DependencyService:Initialize` is getting called.

I think `DependencyService:SetToInitialized()` is getting called before `Initialize()`, and thus the bool is getting set, and we no longer scan the assemblies for DependencyServices.

Maybe https://github.com/dotnet/maui/pull/5174 did this? I see that change added a call to `SetToInitialize()` [here](https://github.com/dotnet/maui/pull/5174/files#diff-dde1fc5ff685106920c096cded904fb3b548e261984e2244e241d8470d276bc3R136).

I wonder if that is a problem... are DependencyServices no longer getting registered by default in this code?

cc @PureWeen @mattleibow

Reply:
We manually register the services now. And the assembly scanning is no longer supported or needs to be explicitly done by the user.

https://github.com/dotnet/maui/blob/main/src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs#L145